src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const sourceObject = {};
+  const declarations = sourceString.split(';');
+
+  declarations.forEach((declaration) => {
+    const [property, value] = declaration.split(':');
+
+    if (property && value) {
+      const trimmedProperty = property.trim();
+      const trimmedValue = value.trim();
 
+      sourceObject[trimmedProperty] = trimmedValue;
+    }
+  });
```

Comment:
The use of 'forEach' here is appropriate as it does not return a new array. However, consider using 'reduce' to build the object directly, which can be more declarative and concise.